### PR TITLE
[Merged by Bors] - feat: the Fourier transform is self-adjoint

### PR DESCRIPTION
**Pull request by sgouezel**
Also extend the notation `𝓕` for the Fourier transform to inner product spaces, introduce a notation `𝓕⁻` for the inverse Fourier transform, and generalize a few results from the real line to inner product spaces.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Fourier/FourierTransform.lean [hunk 1]
```diff
@@ -251,23 +321,71 @@ theorem vector_fourierIntegral_eq_integral_exp_smul {V : Type*} [AddCommGroup V]
...
+  [InnerProductSpace ℝ V] [MeasurableSpace V] [BorelSpace V] [FiniteDimensional ℝ V]
+  {W : Type*} [NormedAddCommGroup W]
+  [InnerProductSpace ℝ W] [MeasurableSpace W] [BorelSpace W] [FiniteDimensional ℝ W]
+
+open scoped RealInnerProductSpace
+
+/-- The Fourier transform of a function on an inner product space, with respect to the standard
+additive character `ω ↦ exp (2 i π ω)`. -/
+def fourierIntegral (f : V → E) (w : V) : E :=
+  VectorFourier.fourierIntegral Real.fourierChar volume (innerₗ V) f w
```
Comment (loefflerd) [inline Mathlib/Analysis/Fourier/FourierTransform.lean:336]:
```suggestion
  VectorFourier.fourierIntegral 𝐞 volume (innerₗ V) f w
```

File: Mathlib/Analysis/Fourier/FourierTransform.lean [hunk 2]
```diff
@@ -251,23 +321,71 @@ theorem vector_fourierIntegral_eq_integral_exp_smul {V : Type*} [AddCommGroup V]
...
 #align real.fourier_integral Real.fourierIntegral
 
-theorem fourierIntegral_def (f : ℝ → E) (w : ℝ) :
+/-- The inverse Fourier transform of a function on an inner product space, defined as the Fourier
+transform but with opposite sign in the exponential. -/
+def fourierIntegralInv (f : V → E) (w : V) : E :=
+  VectorFourier.fourierIntegral Real.fourierChar volume (-innerₗ V) f w
```
Comment (loefflerd) [inline Mathlib/Analysis/Fourier/FourierTransform.lean:342]:
```suggestion
  VectorFourier.fourierIntegral 𝐞 volume (-innerₗ V) f w
```

File: Mathlib/Analysis/Fourier/FourierTransform.lean [hunk 3]
```diff
@@ -251,23 +321,71 @@ theorem vector_fourierIntegral_eq_integral_exp_smul {V : Type*} [AddCommGroup V]
...
+  simp only [fourierIntegral_eq, ofAdd_neg, map_inv, coe_inv_unitSphere, Function.comp_apply,
+    ← MeasurePreserving.integral_comp A.measurePreserving A.toHomeomorph.measurableEmbedding,
+    ← A.inner_map_map]
+
+lemma fourierIntegralInv_comp_linearIsometry (A : W ≃ₗᵢ[ℝ] V) (f : V → E) (w : W) :
+    𝓕⁻ (f ∘ A) w = (𝓕⁻ f) (A w) := by
+  simp [fourierIntegralInv_eq_fourierIntegral_neg, fourierIntegral_comp_linearIsometry]
+
+theorem fourierIntegral_real_eq (f : ℝ → E) (w : ℝ) :
     fourierIntegral f w = ∫ v : ℝ, fourierChar[-(v * w)] • f v :=
```
Comment (loefflerd) [inline Mathlib/Analysis/Fourier/FourierTransform.lean:377]:
```suggestion
    fourierIntegral f w = ∫ v : ℝ, 𝐞[-(v * w)] • f v :=
```

**Review by loefflerd — APPROVED** [review]
Looks great! I like your custom notation `𝐞` as well – the only change I have to suggest is a couple of further places where the notation can be used. It's a trivial thing, so feel free to ignore if you disagree. Either way, definitely a positive review from me.

**Comment by sgouezel** [comment]
Thanks for the review!

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**